app/src/test/java/org/mozilla/fenix/ext/BookmarkNodeTest.kt

```diff
@@ -0,0 +1,41 @@
+package org.mozilla.fenix.ext
+
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.TestApplication
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+import mozilla.components.concept.storage.BookmarkNode
+import mozilla.components.concept.storage.BookmarkNodeType
+
+@RunWith(RobolectricTestRunner::class)
+@Config(application = TestApplication::class)
+class BookmarkNodeTest {
+
+    @Test
+    fun `minus`() {
```

Comment:
This test tests only one case: we're subtracting a subset of the children. However, there are a few other classes of problems:
- The minus children equals the parent's children so they're all removed
- The minus children is a has some parent's children and some not parent's children: ensure only the appropriate ones are removed.
- Minus children is independent of parent's children: nothing is removed
- Parents children is an empty list (expected: no crash & an empty list at the end?)
- Parents children is null (expected: no crash & null at the end?)
- Minus children is an empty set: nothing is removed.

These are potentially all worth testing too. tbh, I'm not sure it's worth the time. In fact, there are even more cases to consider which may or may not be worth the time:
- Does minus ensure the other attributes are unmodified?
- Do the other values in the object (E.g. a null `parentGuid`) modify the results of `minus`?

A big benefit of testing (especially as a dev writing your own tests) is to be forced to think through all of these different "classes of problems" and ensure the implementation is correct for these edge cases. As for what's actually worth writing, I usually start with:
- What you have (i.e. a happy path)
- The edge cases: null state, empty collections, etc.

I'd like to see this here as well :) but it's not strictly necessary: I don't know what else you have to test and so what your priorities are. Note that if you do do this, make sure to leverage helper functions to avoid duplication as much as possible. e.g. because you don't care about all the other values in `BookmarkNode`, you could conceivably have a `newBookmarkNode(name: ..., children...): BookmarkNode` function that greatly simplifies node creation.